### ISSUE TITLE
[BUG] AccessControlException: access denied on start

### ISSUE THREAD

**Issue by wkruse** [opensearch-project/security#2065]
We are running the official `opensearchproject/opensearch:2.2.1` Docker container and getting following exception on start:

```
[2022-09-02T09:14:17,657][ERROR][o.o.s.c.ConfigurationRepository] [opensearch-0] Cannot apply default config (this is maybe not an error!)
com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Failed on call to `getDeclaredMethods()` on class `org.opensearch.common.xcontent.ToXContent`, problem: (java.security.AccessControlException) access denied ("java.lang.RuntimePermission" "accessDeclaredMembers")
	at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:77) ~[jackson-databind-2.13.3.jar:2.13.3]
	at com.fasterxml.jackson.databind.SerializerProvider.reportBadDefinition(SerializerProvider.java:1300) ~[jackson-databind-2.13.3.jar:2.13.3]
	at com.fasterxml.jackson.databind.SerializerProvider._createAndCacheUntypedSerializer(SerializerProvider.java:1447) ~[jackson-databind-2.13.3.jar:2.13.3]
	at com.fasterxml.jackson.databind.SerializerProvider.findValueSerializer(SerializerProvider.java:544) ~[jackson-databind-2.13.3.jar:2.13.3]
	at com.fasterxml.jackson.databind.SerializerProvider.findTypedValueSerializer(SerializerProvider.java:822) ~[jackson-databind-2.13.3.jar:2.13.3]
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:308) ~[jackson-databind-2.13.3.jar:2.13.3]
	at com.fasterxml.jackson.databind.ObjectMapper._writeValueAndClose(ObjectMapper.java:4568) ~[jackson-databind-2.13.3.jar:2.13.3]
	at com.fasterxml.jackson.databind.ObjectMapper.writeValueAsString(ObjectMapper.java:3821) ~[jackson-databind-2.13.3.jar:2.13.3]
	at org.opensearch.security.support.ConfigHelper.createEmptySdcYaml(ConfigHelper.java:115) ~[opensearch-security-2.2.1.0.jar:2.2.1.0]
	at org.opensearch.security.support.ConfigHelper.createFileOrStringReader(ConfigHelper.java:96) ~[opensearch-security-2.2.1.0.jar:2.2.1.0]
	at org.opensearch.security.support.ConfigHelper.uploadFile(ConfigHelper.java:72) ~[opensearch-security-2.2.1.0.jar:2.2.1.0]
	at org.opensearch.security.configuration.ConfigurationRepository$1.run(ConfigurationRepository.java:144) [opensearch-security-2.2.1.0.jar:2.2.1.0]
	at java.lang.Thread.run(Thread.java:833) [?:?]
	Suppressed: org.yaml.snakeyaml.emitter.EmitterException: expected NodeEvent, but got <org.yaml.snakeyaml.events.DocumentEndEvent()>
		at org.yaml.snakeyaml.emitter.Emitter.expectNode(Emitter.java:412) ~[snakeyaml-1.26.jar:?]
		at org.yaml.snakeyaml.emitter.Emitter.access$1600(Emitter.java:62) ~[snakeyaml-1.26.jar:?]
		at org.yaml.snakeyaml.emitter.Emitter$ExpectDocumentRoot.expect(Emitter.java:379) ~[snakeyaml-1.26.jar:?]
		at org.yaml.snakeyaml.emitter.Emitter.emit(Emitter.java:220) ~[snakeyaml-1.26.jar:?]
		at com.fasterxml.jackson.dataformat.yaml.YAMLGenerator._emit(YAMLGenerator.java:969) ~[jackson-dataformat-yaml-2.13.3.jar:2.13.3]
		at com.fasterxml.jackson.dataformat.yaml.YAMLGenerator._emitEndDocument(YAMLGenerator.java:964) ~[jackson-dataformat-yaml-2.13.3.jar:2.13.3]
		at com.fasterxml.jackson.dataformat.yaml.YAMLGenerator.close(YAMLGenerator.java:489) ~[jackson-dataformat-yaml-2.13.3.jar:2.13.3]
		at com.fasterxml.jackson.databind.util.ClassUtil.closeOnFailAndThrowAsIOE(ClassUtil.java:497) ~[jackson-databind-2.13.3.jar:2.13.3]
		at com.fasterxml.jackson.databind.ObjectMapper._writeValueAndClose(ObjectMapper.java:4570) ~[jackson-databind-2.13.3.jar:2.13.3]
		at com.fasterxml.jackson.databind.ObjectMapper.writeValueAsString(ObjectMapper.java:3821) ~[jackson-databind-2.13.3.jar:2.13.3]
		at org.opensearch.security.support.ConfigHelper.createEmptySdcYaml(ConfigHelper.java:115) ~[opensearch-security-2.2.1.0.jar:2.2.1.0]
		at org.opensearch.security.support.ConfigHelper.createFileOrStringReader(ConfigHelper.java:96) ~[opensearch-security-2.2.1.0.jar:2.2.1.0]
		at org.opensearch.security.support.ConfigHelper.uploadFile(ConfigHelper.java:72) ~[opensearch-security-2.2.1.0.jar:2.2.1.0]
		at org.opensearch.security.configuration.ConfigurationRepository$1.run(ConfigurationRepository.java:144) [opensearch-security-2.2.1.0.jar:2.2.1.0]
		at java.lang.Thread.run(Thread.java:833) [?:?]
```

**Comment by peternied** [comment]
It sounds like @cwperks provided a work around, @wkruse can you attempt this?

If that doesn't work please use the support forums https://forum.opensearch.org/c/security/3 for future requests.

**Comment by wkruse** [comment]
Adding `allowlist.yml` fixes the exception on start.

As `allowlist.yml` should replace the `whitelist.yml`, after adding the former we removed the latter. But then we get the following error from `securityadmin.sh`:

```
ERR: Seems ../../../config/opensearch-security/whitelist.yml is not in legacy format: java.io.FileNotFoundException: ../../../config/opensearch-security/whitelist.yml (No such file or directory)
```

So it looks like, OpenSearch server expects having `allowlist.yml`, but `securityadmin.sh` expects having `whitelist.yml`.

Closing it, as the exception on start is fixed for us (by having both of the files).
